crates/core/src/repository/command_input.rs

```diff
@@ -220,7 +221,42 @@ impl OnFailure {
 }
 
 /// helper to split arguments
-// TODO: Maybe use special parser (winsplit?) for windows?
+#[cfg(not(windows))]
 fn split(s: &str) -> RusticResult<Vec<String>> {
     Ok(shell_words::split(s).map_err(|err| RusticErrorKind::Command(err.into()))?)
 }
+
+/// helper to split arguments
+///
+// We keep the return type as `RusticResult<Vec<String>>` to keep the internal api
+// consistent with the other platforms.
+#[allow(clippy::unnecessary_wraps)]
+#[cfg(windows)]
+fn split(s: &str) -> RusticResult<Vec<String>> {
```

Comment:
with this, this is a change and not only a test PR ;-)

I experimented shortly on winsplit. One drawback I found was that it seems to be impossible to have the reverse operation of `split` - this makes a suitable `Display` impl quite hard.

Also I just wanted to note that a OS-dependent parsing may cause irritation for users using both windows and non-windows. I don't use windows, but are there cases where `shellwords::split` don't work on windows commands?

Reply:
Yeah, I'm playing around with it to get a feeling for the parsing. It's true, that we need to be attentive here.